python/ray/tune/callback.py

```diff
@@ -278,12 +283,91 @@ def on_experiment_end(self, trials: List["Trial"], **info):
         """
         pass
 
+    def get_state(self) -> Optional[Dict]:
+        """Get the state of the callback.
+
+        This method should be implemented by subclasses to return a dictionary
+        representation of the object's current state.
+
+        Returns:
+            state: State of the callback. Should be `None` if the callback does not
+                have any state to save (this is the default).
+        """
+        return None
+
+    def set_state(self, state: Dict):
+        """Get the state of the callback.
+
+        This method should be implemented by subclasses to restore the callback's
+        state based on the given dict state.
+
+        Args:
+            state: State of the callback.
+        """
+        pass
```

Comment:
Should we mention here that `CKPT_FILE_TMPL` should be overriden?

Reply:
Individual callbacks can no longer get saved. Can only save the full callback list, so no need to worry about multiple callbacks overwriting each other.